Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -13,58 +13,182 @@
 
 static GLuint s_bbox_buffer_id;
 
+static int s_image_mask_bounds[4] = {0, 0, 0, 0};
+static u8* s_image_mask_data = nullptr;
+static GLuint s_image_mask_texture;
+
+static float s_xscale;
+static float s_yscale;
+
 namespace OGL
 {
-void BoundingBox::Init()
+GLuint BoundingBox::image_mask_fbo;
+bool BoundingBox::image_mask_updated;
+bool BoundingBox::image_mask_cleared;
+
+void BoundingBox::SetEfbScaleChanged(float xscale, float yscale)
 {
-  if (g_ActiveConfig.backend_info.bSupportsBBox)
+  s_xscale = xscale;
+  s_yscale = yscale;
+
+  if (!g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+  {
+    image_mask_updated = false;
+
+    if (s_image_mask_data)
+      free(s_image_mask_data);
+    s_image_mask_data = reinterpret_cast<u8*>(
+        malloc(static_cast<int>(640 * xscale) * static_cast<int>(528 * yscale)));
```

Comment:
`static_cast<int>(640 * xscale)` and `static_cast<int>(528 * yscale)` are repeated quite frequently. You may want to consider local helper functions.